### PR TITLE
Add reproducible building of init and add top-level build infrastructure

### PR DESCRIPTION
**Pull request by foersleo**
**Issue #, if available:** - 

**Description of changes:**

We have been working on reproducible builds of the binaries that go into `aws-nitro-enclaves-cli` for a while now.
This PR provides the last puzzle piece to also convert the build for `init` to the same nix-based approach and ties
everything together at the top level to tell a coherent story of how the binaries are build.

With this change you can choose to either build just parts of the binaries or the whole set either natively through nix
or in a docker container.

The main changes here are in the first and fourth commit with some cleanup and minor adjustments in the remaining
3 commits.

**Testing done:**

Tested building all subvariants in Docker and natively through nix on both supported architectures (aarch64 and x86_64).
Used the resulting set of blobs on both architectures to build and run the example `hello` application from https://github.com/aws/aws-nitro-enclaves-cli/tree/main/examples

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by mariusknaust — DISMISSED** [review]

File: init/init.nix
Comment (eugkoira) [inline init/init.nix:1]: Having a dedicated README for this component similarly to others would be useful

File: README.md [hunk 1]
```diff
@@ -2,38 +2,68 @@
...
 
 ### Prerequisites
 
-The kernel download step requires setting up gpg2 with the kernel developer
-keys. Instructions are available [here](https://www.kernel.org/category/signatures.html).
+The binaries can be build through one of two ways:
+1) Build through [Nix](https://nixos.org/download.html), wich requires a local nix environment installed.
+2) Build through [Docker](https://www.docker.com/), which requires a local docker environment installed.
+
+### Building Docker Nix
```
Comment (eugkoira) [inline README.md:15]: I think you meant here:
```
### Building with Nix
```

File: README.md [hunk 2]
```diff
@@ -2,38 +2,68 @@
...
+### Building with Docker
 
-The project can be built inside a Docker container to avoid installing toolchains and other packages
-on your local device.
+The project can be built inside a Docker container to avoid installing Nix on your local device.
+
+Build the full set of binaries through:
+
+```
+docker build -t blobs_all .
```
Comment (eugkoira) [inline README.md:50]: `docker build -t artifacts`?

File: README.md (hunk 1 above)
Comment (foersleo) [reply README.md:15]: Yes. Will fix.

File: README.md (hunk 2 above)
Comment (foersleo) [reply README.md:50]: It is just assigning a tag (or a list of tags) to the docker image for easier reference (See option `-t, --tag` for `docker build`), so we do not need to look up the image id and use that in the extract phase below.

We would need to specify `artifacts` if we were to use the `--target` option for docker build.

It is just some identifier for easier use, but I am happy to change that to any other tag that people see more fitting.

File: init/init.nix
Comment (foersleo) [reply init/init.nix:1]: I can do that. I was also contemplating about removing the subpackage `README.md`s for linuxkit and kernel as well, as you can now easily just build all this from the top level and there is not necessarily a need to have dedicated readme for each subcomponent. Especially, as the submodule readme files mostly just talk about how to build, which is also done in the top-level one.

Comment (eugkoira) [reply init/init.nix:1]: also fine, the only thing would be great to have is some overview of every component: mostly why it is needed for `nitro-cli` to work

File: README.md (hunk 2 above)
Comment (eugkoira) [reply README.md:50]: sorry, my bad, I confused it with `--target`

File: init/init.nix
Comment (foersleo) [reply init/init.nix:1]: Added extra data on the top-level README and removed the sub-package README files. Let me know what you think.

**Review by eugkoira — APPROVED** [review]

**Review by petreeftime — APPROVED** [review]

**Comment by foersleo** [comment]
Found a minor detail in the readme where in the docker example I had `--build-args` where it should have been `--build-arg`. Going to override merge this as this is merely a cosmetic issue in the documentation.

**Pull request merged**